File: backend/app/src/main/java/com/dictionary/app/Models/Word.java
package com.dictionary.app.Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "words")
public class Word {
    @Id
    @GeneratedValue
    private Integer id;

    @Column(nullable = false)
    private String word;

    @Column
    private String definition;

    /*
    @Column
    private String phrase;

     */

    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "root_id", nullable = false)
    private WordRoot root;

    @Column
    private String audioFile;


}

File: frontend/DictionaryApp/DictionaryApp/Models/Word.cs
using DictionaryApp.Models;
using Newtonsoft.Json;

namespace DictionaryApp
{
    public class Word
    {
        [JsonProperty("id")]
        public int id { get; set; }

        [JsonProperty("wordName")]
        public string wordName { get; set; }

        [JsonProperty("definition")]
        public string? definition { get; set; }

        [JsonProperty("audioFile")]
        public string? audioFile { get; set; }

        [JsonProperty("root")]
        public WordRoot Root { get; set; }
    }
}